pubsub/api/Pubsub.Samples.Tests/PubsubFixture.cs

```diff
@@ -0,0 +1,132 @@
+// Copyright 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Cloud.PubSub.V1;
+using GoogleCloudSamples;
+using Grpc.Core;
+using System;
+using System.Collections.Generic;
+using Xunit;
+
+[CollectionDefinition(nameof(PubsubFixture))]
+public class PubsubFixture : IDisposable, ICollectionFixture<PubsubFixture>
+{
+    public readonly string ProjectId;
+    public List<string> TempTopicIds { get; set; } = new List<string>();
+    public List<string> TempSubscriptionIds { get; set; } = new List<string>();
+
+    public PubsubFixture()
+    {
+        ProjectId = Environment.GetEnvironmentVariable("GOOGLE_PROJECT_ID");
+    }
+
+    public void Dispose()
+    {
+        var deleteTopicSampleObject = new DeleteTopicSample();
+        var deleteSubscriptionSampleObject = new DeleteSubscriptionSample();
+        foreach (string subscriptionId in TempSubscriptionIds)
+        {
+            _retryRobot.Eventually(HandleDeleteRace(() =>
+                deleteSubscriptionSampleObject.DeleteSubscription(ProjectId, subscriptionId)));
+        }
+        foreach (string topicId in TempTopicIds)
+        {
+            _retryRobot.Eventually(HandleDeleteRace(() =>
+                deleteTopicSampleObject.DeleteTopic(ProjectId, topicId)));
+        }
+    }
+
+    private readonly RetryRobot _retryRobot = new RetryRobot()
```

Comment:
I think this can go as well, you are using the samples to delete and they wait for the resource to be deleted, we are not doing anyghing concurrently.